shared/test/loop_mocha_utils.js

```diff
@@ -43,8 +43,18 @@ var LoopMochaUtils = (function(global, _) {
         return this;
       }
 
-      continuations.push(contFn);
-      return this;
+      // Allow chaining of `then` by returning Promises that resolve on result.
+      return new SyncThenable(resolve => {
+        continuations.push(result => {
+          let ret = contFn(result);
+          if (ret instanceof SyncThenable) {
+            ret.then(resolve);
+          }
+          else {
+            resolve(ret);
+          }
+        });
+      });
```

Comment:
It'd be nice to make this a bit more readable on first approach.  It might help to make the outer arrow function a separate function, though I kinda suspect not.  Maybe just another comment or two as to how the mechanics work?